src/dlr_tensorflow/dlr_tensorflow.cc

```diff
@@ -87,6 +87,84 @@ TF_Output TensorflowModel::ParseTensorName(const std::string& t_name) {
   return oper_out;
 }
 
+void TensorflowModel::DetectInputs() {
+  size_t pos = 0;
+  TF_Operation* op;
+  while ((op = TF_GraphNextOperation(graph_, &pos)) != nullptr) {
+    const std::string op_type = TF_OperationOpType(op);
+    const int n_in = TF_OperationNumInputs(op);
+    const int n_out = TF_OperationNumOutputs(op);
+    const std::string op_name = TF_OperationName(op);
+    if (op_type == "Placeholder" && n_in == 0 && n_out == 1) {
+      input_names_.push_back(op_name + ":0");
+    }
+  }
+  num_inputs_ = input_names_.size();
+}
+
+void TensorflowModel::DetectOutputs() {
+  size_t pos = 0;
+  TF_Operation* op;
+  // while loop
+  while ((op = TF_GraphNextOperation(graph_, &pos)) != nullptr) {
+    const std::string op_type = TF_OperationOpType(op);
+    const int n_out = TF_OperationNumOutputs(op);
+    const int n_cout = TF_OperationNumControlOutputs(op);
+    const std::string op_name = TF_OperationName(op);
+    if (op_type != "Const" && op_type != "Assign" && op_type != "NoOp" &&
+        op_type != "Placeholder" && n_cout == 0) {
+      int n_cons = 0;
```

Comment:
Since there are already a few `n_*` variables going around here, it may be helpful to change `n_cons` to `n_consumers` to be more clear.

Reply:
fixed